### DISCUSSION TITLE
When you move the window, the frame inside is also offset, and jitter will occur.

### DISCUSSION THREAD

**Discussion by han1548772930** [fltk-rs/fltk-rs#1559] [Q&A]
```rust
   if let Some(win) = area_win.as_mut() {
                win.resize(left, top, right - left, bottom - top);
                if let Some(f) = area_frame.as_mut() {
                    f.set_pos(-left, -top);
                }
            }
```
Is there any better approach to this situation?

**Comment by han1548772930** [comment]
Or is it possible to make the window fully transparent with some grayscale?

**Comment by MoAlyousef** [comment]
It’s difficult to say from that code snippet. If you call resize inside a resize_cb it might cause some recursion. 
The best way is to let FLTK handle positioning for you. Usually you would use a flex widget (or several) to organize your children widgets. This should make resizing automatic and without any jitters. 


fltk-rs windows have a set_opacity method which allows to set the transparency of the window.

**Reply by han1548772930** [reply]
Is there any way to offset it based on the lower right corner?

**Reply by MoAlyousef** [reply]
Flex rows start from the left. You can fill it with an empty Frame then place a column for example to the right. The column itself starts from the top, so you can also fill it with an empty frame and place a widget at the bottom.
If you want to use coordinates, you would have to do your own calculations to get the lower right corner.
An example of Flexes:
```rust
use fltk::{app, prelude::*, group::Flex, frame::Frame, button::Button, window::Window};


fn main() {
    let a = app::App::default();
    let mut window = Window::default().with_size(320, 240);
    let mut row = Flex::default_fill().row();
    Frame::default();
    let mut col = Flex::default().column();
    Frame::default();
    let mut button = Button::default().with_label("Click");
    col.fixed(&button, 30);
    col.end();
    row.fixed(&col, 80);
    row.end();
    window.show();

    a.run().unwrap();
}
```
![image](https://github.com/fltk-rs/fltk-rs/assets/37966791/354a0bee-5658-4f30-8c49-34cf78ba125a)

Another option is to use the fltk-grid crate.
